### PR TITLE
Bb/bootstrap integration

### PR DESCRIPTION
**Pull request by bbland1**
## Description

We discussed using `bootstrap` and `react-bootstap` in the application this installs them and changes our buttons, inputs, navbar to utilize it. Makes sure the app still looks and functions the same with the `bootstrap` default styling basically.

## Acceptance Criteria

- [x] A working and functioning app that uses the new package 😅

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓    | 💄  Style |

### Before

Screenshot 2024-09-25 at 7 06 50 PM

### After

Screenshot 2024-09-25 at 7 06 59 PM

## Testing Steps / QA Criteria

- Opening the [preview link]() in the PR you should see the changes

To make sure the dependencies and code function as should:
- From your terminal, pull down this branch with `git pull origin bb/bootstrap-integration` and check that branch out with git checkout `bb/bootstrap-integration``
- Then `npm ci` to install the newly added dependencies locally and `npm start` to launch the app.
- Open a second terminal to run the typescript compiler using ` tsc --noEmit --watch` to see any type issue.
- After installing the new dependencies and starting the app you should see that default styling of the `bootstrap` on all buttons and inputs
- Navigate through the application making sure all functionality still works

### CODE REVIEW

**Comment by zahrafalak** [comment]
This works great! Thanks for integrating it in! I can see the default styles (that we can alter later on ofc), and didn't notice anything breaking. 🚀

File: index.html [hunk 1]
```diff
@@ -11,6 +11,12 @@
 			content="A smart shopping list that learns your purchase habits and makes suggestions, so you don't forget to buy what's important."
 		/>
 		<link rel="icon" type="image/svg+xml" href="/src/favicon.ico" />
+		<link
+			rel="stylesheet"
+			href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css"
+			integrity="sha384-T3c6CoIi6uLrA9TneNEoa7RxnatzjcDSCmG1MXxSR1GAsXEV/Dwwykc2MPK8M2HN"
+			crossorigin="anonymous"
+		/>
```
Comment (zahrafalak) [inline index.html:19]: This works great! Thanks for integrating it in! I can see the default styles (that we can alter later on ofc), and didn't notice anything breaking. 🚀

Comment (bbland1) [reply index.html:19]: Yeah, I wanted tojust get it and and using it to make sure that 
1. it worked and didn't break everything 😂😅
2. we would still have options for css (like the sass suggestion you made!)

File: src/api/useAuth.tsx [hunk 2]
```diff
@@ -3,27 +3,28 @@ import { auth } from "./config.js";
...
  */
 export const SignInButton = () => (
-	<button
+	<Button
 		type="button"
 		onClick={() => signInWithPopup(auth, new GoogleAuthProvider())}
 	>
 		Sign In
-	</button>
+	</Button>
```
Comment (RossaMania) [inline src/api/useAuth.tsx:19]: Implementing that Bootstrap Button component! That Button Bootstrap component? Implementing that Bootstrap component! Nice!

File: src/components/AuthenticatedNavBar.tsx [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+import Navbar from "react-bootstrap/Navbar";
+import Container from "react-bootstrap/Container";
 
 import "./AuthenticatedNavBar.css";
 
 export function AuthenticatedNavBar() {
 	return (
-		<nav className="Nav">
-			<div className="Nav-container">
+		<Navbar className="Nav">
```
Comment (RossaMania) [inline src/components/AuthenticatedNavBar.tsx:11]: Writing this here, so I hopefully don't forget: There should be a Navlink Container Bootstrap component. A good "tomorrow problem"!

File: src/components/FilterListInput.tsx [hunk 4]
```diff
@@ -17,19 +20,24 @@ export function FilterListInput({
...
+	};
+
 	return (
-		<form onSubmit={handleSubmit}>
-			<label htmlFor="filterList">
-				Filter List:
-				<input
-					type="search"
+		<Form onSubmit={handleSubmit}>
+			<InputGroup>
```
Comment (RossaMania) [inline src/components/FilterListInput.tsx:29]: Nice use of that Form Bootstrap component!

**Review by RossaMania — APPROVED** [review]
Great implementation of Bootstrap! You were able to install and import Bootstrap, and use some default Bootstrap components without making big styling decisions! Nice job! This should give us some inspiration

**Comment by eternalmaha** [comment]
Thanks for implementing this! The blue and more animated feel of the buttons in the list view is growing on me. Everything appears to be working as it should too!!! The only odd thing I noticed (and this probably existed beforehand  since I don't see how bootstrap would change this unless its a default sizing thing built in, i'm not really sure honestly?) For my absurdly large list, the list is cut off at when my window is minimized somewhat. Im not able to scroll down to the bottom of the list. The issue goes away and I can scroll normally at full window. This could probably be addressed later when we make styling changes that'll improve responsiveness i'm sure too!!! Great job!!! Thanks for doing this and getting us started!!! Screenshot 2024-09-27 at 4 15 50 PM

File: src/components/AuthenticatedNavBar.tsx (hunk 3 above)
Comment (bbland1) [reply src/components/AuthenticatedNavBar.tsx:11]: Great point 😅 I was nervous to mess with the navbar too much it was late and I couldn't wrap my head aorund it pass the Navbar and container!

File: src/views/authenticated/List.tsx [hunk 5]
```diff
@@ -53,26 +54,26 @@ export function List({ data: unfilteredListItems, listPath }: Props) {
...
-					)}
-					<h3>Want to add more items to your list?</h3>
-					<button
-						onClick={() => navigate("/manage-list")}
-						aria-label="Navigate to add more items to your list"
-					>
-						{"Add items"}
-					</button>
-				</section>
+			<section className="sticky-top bg-dark">
```
Comment (bbland1) [inline src/views/authenticated/List.tsx:57]: This is not the right background color but needed to set it to something to see the scrolling and make sure ti stuck to the top correctly!

**Review by alex-andria — APPROVED** [review]
Nice work Brianna and thanks for taking initiative on this! Already looking great and can't wait to see how you all style this with the SCSS PR Falak put up as well.

**Pull request merged**